Sources/Apodini/SyntaxTreeVisitor/SyntaxTreeVisitor.swift

```diff
@@ -112,7 +112,7 @@ class SyntaxTreeVisitor {
     
     private func formHandlerIndexPathForCurrentNode() -> HandlerIndexPath {
         let rawValue = currentNodeIndexPath
-            .map { String($0 - 1) } // We remove one from the current indexPath to have 0 as the first index
+            .map { String(max($0, 1) - 1) } // We remove one from the current indexPath to have 0 as the first index
```

Comment:
Would you mind explaining why this needs to be changed? Or why the change in Group resulted in this change to be necessary 🤔
This max operation just seems a bit odd to me as it is not injective. But the test are seemingly passing so might just be my missing knowledge.

Reply:
There was an issue w/ one of the tests where the handler index path would contain -1, that's why i added the `max` call back in.

It is, in fact, not needed. (And now removed)

The issue seems to be the following: In some of the tests we generate the web service model the following way:
```swift
let builder = SharedSemanticModelBuilder(app).with(exporter: RESTInterfaceExporter.self)
let visitor = SyntaxTreeVisitor(semanticModelBuilders: [builder])
TestWebService().accept(visitor)
visitor.finishParsing()
```

Line 3 here gets resolved to [`Component.accept`](https://github.com/Apodini/Apodini/blob/b5b4be3c9a0f48748b85154d196257bdd76543d6/Sources/Apodini/Components/Component.swift#L61), which is not what we want (because, unlike [`WebService.visit`](https://github.com/Apodini/Apodini/blob/b5b4be3c9a0f48748b85154d196257bdd76543d6/Sources/Apodini/WebService.swift#L83), it doesn't insert the implicit root `Group`, and also doesn't set the version context).

We should be using [`WebService.register`](https://github.com/Apodini/Apodini/blob/b5b4be3c9a0f48748b85154d196257bdd76543d6/Sources/Apodini/WebService.swift#L77), since that properly sets up the root context.

Fun fact: this probably also means that some of the other tests which still use `someWebService.accept(visitor)` rather than `WebService.register` probably test against generated endpoints (since they all are missing the root context).

(You can see this in [this commit](https://github.com/Apodini/Apodini/pull/120/commits/ad7dc19b78b43d6b4e5fefad43f5d5e842dd8971): the test passed w/ both the old and the new version, because the .accept version is missing the root level)

And the missing root level (alongside the then also missing implicit root Group) caused the index path indices to become negative (which the max call fixed).

Interestingly enough the issue w/ negative indices could only ever have occurred in our tests, since the `WebService.accept` function is private. Also means that the max call, when reached for a user-defined web service, always was effectively a no-op since `$0` always was >= 1.

Reply:
Thanks for the detailed explanation 🚀 . So the real difference is the missing `Group` Component inserted by the `WebService` by default? Ah I see, if the `TestWebService` would just be a simple `Text` component for example, the `ContextNode` wouldn't be properly prepared, as there was not call to `enterContent` right?
Would it make sense to prepare the `SyntaxTreeVisitor` in such a way, such scenarios are handled as well? Not sure if one might remove the `Group` in the `WebService` in the future? If I remember correctly there was a time where `currentNodeIndexPath` hold a `0` by default right?

Reply:
Yeah pretty much. We used to have `currentNodeIndexPath = [0]` as the default value, but that was removed since it wasn't really needed.

We can move some of this into the visitor, but that'd mean we have to make sure that the visitor isn't reused (which would be a good idea anyway, since it shouldn't).
Also would mean that, while we'd then support WebServices w/ top-level non Group/Tuple components (see other comment below), other WebServices with such top-level components would have too many entries in the `currentNodeIndexPath` array (since we'd have both the visitor-added root context as well as the one introduced by the root component).

Reply:
> [...] we have to make sure that the visitor isn't reused [...]

Think this would be a good addition. We are able to determine if parsing is complete afaik, we should be able to simple flip a bool and refuse any more calls to the SyntaxTreeVisitor 🤔 Or any other way which is maybe more "efficient".

> [...] other WebServices with such top-level components would have too many entries in the currentNodeIndexPath array (since we'd have both the visitor-added root context as well as the one introduced by the root component).

Is there a way to detect if we have such a root Group? (Or TupleComponenet, I think the same thin applies there right?)

This would allow removing the root Group in the WebService right? (If I followed the discussion below correctly).